server/src/main/java/org/elasticsearch/transport/SniffConnectionStrategy.java

```diff
@@ -242,7 +243,11 @@ private void collectRemoteNodes(Iterator<Supplier<DiscoveryNode>> seedNodesSuppl
                     }
                 }
                 logger.warn(() -> "fetching nodes from external cluster [" + clusterAlias + "] failed", e);
-                listener.onFailure(e);
+                if (seedNodesSuppliers.hasNext()) {
+                    listener.onFailure(getNoSeedNodeLeftException(Set.of(e)));
+                } else {
+                    listener.onFailure(e);
```

Comment:
Went back to the original [thread](https://github.com/elastic/elasticsearch/pull/94998#discussion_r1156750443) and I see the intention now: if we hit an exception that we think we can't recover from by moving on to a different seed node, we don't want to wrap in `NoSeedNodeLeftException` here.

Given that, the behavior is still confusing though: suppose we have two seed nodes, and we fail with a `ConnectTransportException` on the first one, then with a "terminating" exception on the second node. In that case, we don't wrap the exception. However, if we had three seed nodes, and failed on the second with a "terminating" exception, we _would_ wrap. 

Sorry if I'm missing something here! If I am, it would be great to include a comment in the code with a little more context. 

As a side note: not wrapping on "terminating" exceptions is what seems to produce the inconsistency in status codes between proxy mode and sniff mode (e.g., for the license failures, for get a 500 in proxy mode, but a 403 in sniff mode). I'm wondering if we should introduce a second class of failure that still produces a 500 for the case where we don't want to return a `NoSeedNodeLeftException`.